src/teleport.cpp

```diff
@@ -20,11 +20,11 @@ bool teleport::teleport( Creature &critter, int min_distance, int max_distance,
         return false;
     }
 
-    const bool c_is_u = &critter == &g->u;
     player *const p = critter.as_player();
+    const bool c_is_u = p ? c_is_u = p == g->u : false;
```

Comment:
Can it simply be
```suggestion
    const bool c_is_u = p == &g->u;
```

Reply:
Yep, better in several ways.

Reply:
Whoops, except it really wants a null check in there to assert that c_is_u cannot be true when p is null.